src/main/java/use_case/probability/stand/ProbabilityStandInteractor.java

```diff
@@ -0,0 +1,237 @@
+    public int standProbability() {
+
+        int score = this.handScore(userCards);
+
+        // User hits a blackjack and automatically wins, our simulator does not take into account for Blackjack ties
+        // with the dealer.
+        if (score == BLACKJACK) {
+            return 100;
+        }
+
+        // User busts and immediately loses.
+        if (score > BLACKJACK) {
+            return 0;
+        }
+
+        //enters here if and only if 0 <= score < BLACKJACK
+        return this.winPercentage(score);
+    }
+
+    /**
+     * The score of your current hand. If you have aces in your hand, handScore will return the highest score possible
+     * before any possible bust (greater than 21). If the hand cannot be under 21 this handScore will return the
+     * smallest possible score.
+     * @return an integer representing the score of your current hand.
+     */
+    private int handScore (Map<Integer, Integer> hand) {
+
+        int result = 0;
+        int aceCount = hand.get(11);
+
+        for (int key : hand.keySet()) {
+            result += hand.get(key) * key;
+            }
+
+        // If the player has a hand with aces and have bust (using any aces as a score of 11), reduce the score by 10
+        // until either all aces have been evaluated as score 1 or the hand is scored to be less than or equal to 21.
+        while (aceCount > 0) {
+            if (result > BLACKJACK) {
+                result -= 10;
+                aceCount--;
+            }
+            else {
+                aceCount = 0;
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Counts the amount of times userDeck wins a game of Blackjack versus any possible version of dealerDeck.
+     * Since a card of the dealerDeck is hidden, winPercentage will condition the probability of each win on
+     * each card in unknownCards. If that card results in a handScore less than 17 another card will be pulled,
+     * meaning that every card in unkownDeck that has not been pulled yet in each test will be pulled. Once counted
+     * winPercentage will divide the amount of winning scenarios by the total amount of scenarios.
+     * @param userScore is the handScore of a player that is an integer element of [0, 20]
+     * @return an int representing the floor of the percentage of winning scenarios userDeck beats dealerDeck.
+     */
+    private int winPercentage(int userScore) {
+
+        Map<String, Integer> winScenario = this.countWinsandGames(userScore, dealerCards, unknownCards);
+
+        return Math.floorDiv(winScenario.get(WINS), winScenario.get(SCENARIO));
+    }
+
+    /**
+     * Actively counts the amount of times userDeck wins a game and builds ALL possible decks the dealer can feasibly
+     * have knowing that the dealer will only ever pull another card from the deck if their score is less than 17.
+     * To take care of a very unlikely edge case, if the deck of cards runs out before the dealer hits 21, this is
```

Comment:
You don't have to worry about this since the deck of cards won't ever run out with four players since the largest hand is eleven cards (AAAA2222333), so you can't have more than 44 cards drawn in a game

Reply:
True, I was just worried if we also had 2 cpus that could also have like a few cards of their own I do realize that the deck will never actually be empty